lib/DDG/Spice/IsItUp.pm

```diff
@@ -8,21 +8,32 @@ use Net::IDN::Encode qw(domain_to_ascii domain_to_unicode);
 use Net::Domain::TLD qw(tld_exists);
 use Data::Validate::IP qw(is_ipv4);
 
-triggers query_lc => qr/^((?:is\s|))(?:https?:\/\/)?([\p{Alnum}\-]+(?:\.[\p{Alnum}\-]+)*?)(?:(\.\pL{2,})|)\s(?:up|down|working|online|status)\?*$/i;
+triggers start => 'isitup', 'isitdown', 'is it up', 'is it down', 'status of';
+triggers end => 'up', 'down', 'working', 'online', 'status', 'up right now';
 
 spice to => 'https://isitup.org/$1.json?callback={{callback}}';
 
 spice proxy_cache_valid => "418 1d";
 
-handle matches => sub {
+my $query_start_regex = qr/(?:is|isitup|isitdown|is it up|is it down|status of|)/;
+my $query_end_regex   = qr/(?:up|down|working|online|status|up right now|)/;
+my $url_regex         = qr/(?:https?:\/\/)?([\p{Alnum}\-]+(?:\.[\p{Alnum}\-]+)*?)(?:(\.\pL{2,})|)/;
 
-    my ($domain, $ascii);
+my $query_regex = qr/^($query_start_regex)\s?$url_regex\s?($query_end_regex)\?*$/i;
+
+handle query_lc => sub {
+
+    my ($domain, $ascii, $root_url);
     my $publicSuffix = Domain::PublicSuffix->new();
 
-    my $root_url = $_[1];
+    my $query = $_;
+
+    $query =~ $query_regex;
+
+    $root_url = $2;
```

Comment:
@moollaza Thanks a lot for pointing it out. I'll make the change.

Can you please share which queries gave this error? I saw it on `is .com up`, did you see the error on a similar query?